http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2StreamHandler.java

```diff
@@ -250,6 +251,16 @@ public void onData(Stream stream, DataFrame frame, Callback callback) {
 
   }
 
+  @Override
+  public void onReset(Stream stream, ResetFrame frame) {
+    result.setResponseCode(String.valueOf(frame.getError()));
+    result.setResponseMessage(frame.toString());
+    result.sampleEnd();
+    result.setSuccessful((result.getResponseCode()=="0")||(result.getResponseCode()=="8"));
```

Comment:
isn't intellij showing a warning here? this kind of comparison is not safe, since strings are objects, and two strings with same text might not be the same instance (remember that == is used to compare object instances, not semantic equality of objects). To compare string equality (as with other objects) use `equals` method . In this case is better to just compare frame.getError() and to not use "magic numbers" compare against `ErrorCode.NO_ERROR.code` and `ErrorCode.CANCEL_STREAM_ERROR.code`